taf/testlib/common/loops.py

```diff
@@ -0,0 +1,303 @@
+"""
+@copyright Copyright (c) 2017, Intel Corporation.
+
+This program is free software; you can redistribute it and/or modify it
+under the terms and conditions of the GNU Lesser General Public License,
+version 2.1, as published by the Free Software Foundation.
+
+This program is distributed in the hope it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for
+more details.
+
+@file  loops.py
+
+@summary  flexible and extensible conditional loops helpers.
+"""
+
+import time
+from contextlib import suppress
+from collections import OrderedDict
+
+
+class CondLoopException(Exception):
+    pass
+
+
+class CondLoopBase(object):
+    """
+    """
+    END_FAIL = False
+    END_SUCCESS = True
+    KEEP_LOOPING = None
+
+    def __init__(self, end_fail=False, end_success=True, keep_looping=None):
```

Comment:
I appreciate that we have class attributes for the different states/results, but then we don't use them in the __init__ as defaults, please:

`def __init__(self, end_fail=END_FAIL, end_success=END_SUCCESS, keep_looking=KEEP_LOOKING):`